domain/application/service/application.go

```diff
@@ -562,14 +556,16 @@ func (s *Service) AddUnits(ctx context.Context, name string, units ...AddUnitArg
 		args[i] = arg
 	}
 
-	err = s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
-		appID, err := s.st.GetApplicationID(ctx, name)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		return s.st.AddUnits(ctx, appID, args...)
-	})
-	return errors.Annotatef(err, "adding units to application %q", name)
+	appUUID, err := s.st.GetApplicationIDByName(ctx, appName)
+	if err != nil {
+		return internalerrors.Errorf("getting application %q id: %w", appName, err)
+	}
+
+	err = s.st.AddUnits(ctx, appUUID, args...)
```

Comment:
This comment might come as naive (and it was probably already discussed outside this patch), shouldn't we have only one state method `AddUnitsByApplicationName(ctx, appName, args...)` (function name to be defined :rofl: ). The `AddUnits` state method is only used here so I don't see why we should keep it with a UUID as param.

Reply:
This is my preferred pattern, after chatting with Joe. Read transactions don't block, so there is no overhead to them really, we're fine to compose them. Combined with the fact app name and id don't change, we're fine to compose them in the service.

Doing so makes be our state methods slightly simpler